Assets/BossRoom/Scripts/Shared/Game/State/GameStateBehaviour.cs

```diff
@@ -53,6 +55,22 @@ public virtual bool Persists
         /// </summary>
         private static GameObject s_ActiveStateGO;
 
+        protected DIScope m_Scope;
```

Comment:
we probably should make this public {get; protected set} parameter - Fernando will need access to this reference for test runner stuff

Reply:
made the set private since at the moment it is only set here in Awake. We can still change that later if we find that a child class needs to set it itself